Automattic/wp-calypso
# Monetize: Move plans to settings screen

## Proposed Changes

* Moves the Monetize > Plans list from its own sub-page to the Settings screen directly.
* Updates all links to the Plans screen to use the new url
* Puts in place a redirect to catch any links to this screen from elsewhere
* We need to update backend links to this screen, which I'm doing in a separate Jetpack PR here: https://github.com/Automattic/jetpack/pull/34316

**Plans Before**
<img width="1077" alt="plans-before" src="https://github.com/Automattic/wp-calypso/assets/21228350/837b24e1-77e5-4b4f-b519-9ddd4fdfc0ab">

**Plans After**
<img width="1079" alt="plans-after" src="https://github.com/Automattic/wp-calypso/assets/21228350/3166b5b6-79f0-4a11-9896-111a4319cbfc">

**Plans After (With No Plans Created Yet)**
<img width="1075" alt="plans-after-with-no-plans" src="https://github.com/Automattic/wp-calypso/assets/21228350/114da94f-2bb2-497a-9f87-dd14cf1d4193">

## Testing Instructions

1. You will need a simple test site with Stripe connected. 
2. Go to `http://calypso.localhost:3000/earn/payments/YOURSITESLUG`
   - If you have not created plans, you should see a screen like the third screenshot above.
   - If you have created plans, you should see a screen like the second screenshot above. 
   - In both cases, you should no longer have to click through to a separate screen to see plans. 
3. Other tests
   - Click the button to add a plan, add a plan, save, and confirm the new plan is added and shows.
   - Go to `http://calypso.localhost:3000/earn/payments/YOURSITESLUG#add-new-payment-plan` and confirm that the plans modal opens immediately. Add a plan and confirm it shows in the list. 
   - Go to `http://calypso.localhost:3000/earn/payments/YOURSITESLUG#add-newsletter-payment-plan` and confirm the plans modal opens immediately - but with the newsletter tier toggle turned on. 
   - Go to the old link at `http://calypso.localhost:3000/earn/payments-plans/YOURSITESLUG` and confirm the url redirects to the new url. 

## Pre-merge Checklist
